### PR TITLE
a8n: Deprecate CampaignPlan.changesets in favor of CampaignPlan.changesetPlans

### PR DESCRIPTION
**Pull request by mrnugget**
This is a follow-up to #7965 and stacked on top of it. See motivation/reasoning in #7965.

I propose the change here because the returned type is a `ChangesetPlansConnection` and not `Changesets`. The same field is called `changesetPlans` on `Campaign`.

**Update**: Decided to only deprecate the field, since it's heavily used the `src` CLI which is now being used by customers.

### CODE REVIEW

**Comment by eseliger** [comment]
Good idea, I’ll change the frontend

**Comment by felixfbecker** [comment]
For context on this, the reason these are the same name because `Changeset` was supposed to be an interface implemented by both `ChangesetPlan` and `ExternalChangeset`, so both the real `Campaign` and the `CampaignPlan` was supposed to have a list of `Changeset`s (the interface).

**Comment by mrnugget** [comment]
Update on this PR:

I'm unsure now about whether to merge this or not, since it would be a breaking change. That matters for the `src` CLI, which we want to ship to customers this week. If we merge this change here, we would need to make `src` backwards compatible.

**Comment by eseliger** [comment]
Hmm 🤔 If there is a `src` version published that would be affected by this change, I think it makes sense to keep it working for a while. We could introduce it as an alias though and deprecate it(?) 
In general, do we have a plan on how to ensure a `src` version from 1.5yr ago doesn't behave weird/break with ugly errors? Maybe a `X-Sourcegraph-Version: 3` header or something and a check that invalidates old `src` versions? 🤔 (Increase to 4 in this PR then, for example)

**Comment by mrnugget** [comment]
> Hmm 🤔 If there is a `src` version published that would be affected by this change, I think it makes sense to keep it working for a while. We could introduce it as an alias though and deprecate it(?)

There's no published version of `src` that uses this, *but* one customer is highly interested using the new `src` version to run actions/create campaign plans/create campaigns and they're on `3.12.x`, so we should try to keep it backwards compatible.

It's a special case, but I'd rather have _this_ customer use this ASAP then to make this change here.

I'll try to see if I can alias it.

**Comment by mrnugget** [comment]
> In general, do we have a plan on how to ensure a `src` version from 1.5yr ago doesn't behave weird/break with ugly errors? Maybe a `X-Sourcegraph-Version: 3` header or something and a check that invalidates old `src` versions? 🤔 (Increase to 4 in this PR then, for example)

See: https://github.com/sourcegraph/src-cli/pull/86

**Comment by mrnugget** [comment]
I've updated the changes so that `CampaignPlan.changesets` is only deprecated. Everyone that reviewed: please take another look.

File: cmd/frontend/graphqlbackend/schema.graphql [hunk 1]
```diff
@@ -482,9 +482,14 @@ type CampaignPlan implements Node {
     # The progress status of generating changesets.
     status: BackgroundProcessStatus!
 
-    # The changesets that will be created by the campaign.
+    # DEPRECATED: This field will be removed in 3.15. Please use changesetPlans instead.
+    # The proposed patches ("plans") for the changesets that will be created by the campaign.
     changesets(first: Int): ChangesetPlanConnection!
 
+    # DEPRECATED: This field will be removed in 3.15. Please use changesetPlans instead.
```
Comment (eseliger) [inline cmd/frontend/graphqlbackend/schema.graphql]: I think this description is wrong

File: cmd/frontend/graphqlbackend/schema.graphql [hunk 2]
```diff
@@ -482,9 +482,14 @@ type CampaignPlan implements Node {
...
+    # The proposed patches ("plans") for the changesets that will be created by the campaign.
+    changesetPlans(first: Int): ChangesetPlanConnection!
```
Comment (felixfbecker) [inline cmd/frontend/graphqlbackend/schema.graphql]:
```suggestion
    changesetPlans(first: Int): ChangesetPlanConnection! @deprecated(reason: "This field will be removed in 3.15. Please use changesetPlans instead.")
```

Comment (eseliger) [reply cmd/frontend/graphqlbackend/schema.graphql]: wrong property :)

**Comment by eseliger** [comment]
removing all references from the frontend now

**Comment by eseliger** [comment]
done, won't approve my own changes though, @sourcegraph/automation-web please someone take a look 😄

**Pull request merged**